tests/python/relax/test_op_binary.py

```diff
@@ -0,0 +1,135 @@
+    assert relax.op.subtract(x, y).op == Op.get("relax.subtract")
+    assert relax.op.multiply(x, y).op == Op.get("relax.multiply")
+    assert relax.op.divide(x, y).op == Op.get("relax.divide")
+    assert relax.op.floor_divide(x, y).op == Op.get("relax.floor_divide")
+    assert relax.op.less(x, y).op == Op.get("relax.less")
+
+
+def _check_inference(bb: relax.BlockBuilder, call: relax.Call, expected_sinfo: relax.StructInfo):
+    ret = bb.normalize(call)
+    tvm.ir.assert_structural_equal(ret.struct_info, expected_sinfo)
+
+
+def test_infer_struct_info_binary_arith():
+    bb = relax.BlockBuilder()
+    x0 = relax.Var("x", R.Tensor((2, 3), "float32"))
+    x1 = relax.Var("x", R.Tensor((1, 3), "float32"))
+    x2 = relax.Var("x", R.Tensor((3, 2, 3), "float32"))
+    x3 = relax.Var("x", R.Tensor((3, 1, 3), "float32"))
+    x4 = relax.Var("x", R.Tensor("float32", ndim=2))
+    y0 = relax.Var("y", R.Tensor((2, 3), "float32"))
+    y1 = relax.Var("y", R.Tensor((4, 3, 2, 1), "float32"))
+    y2 = relax.Var("y", R.Tensor("float32", ndim=2))
+    y3 = relax.Var("y", R.Tensor("float32", ndim=-1))
+
+    _check_inference(bb, relax.op.add(x0, y0), relax.TensorStructInfo((2, 3), "float32"))
+    _check_inference(bb, relax.op.subtract(x1, y0), relax.TensorStructInfo((2, 3), "float32"))
+    _check_inference(bb, relax.op.multiply(x1, y1), relax.TensorStructInfo((4, 3, 2, 3), "float32"))
+    _check_inference(bb, relax.op.divide(x2, y2), relax.TensorStructInfo((3, 2, 3), "float32"))
+    _check_inference(
+        bb, relax.op.floor_divide(x3, y2), relax.TensorStructInfo(dtype="float32", ndim=3)
+    )
+    _check_inference(bb, relax.op.add(x4, y0), relax.TensorStructInfo((2, 3), dtype="float32"))
+    _check_inference(bb, relax.op.add(x4, y1), relax.TensorStructInfo(dtype="float32", ndim=4))
+    _check_inference(bb, relax.op.add(x4, y2), relax.TensorStructInfo(dtype="float32", ndim=2))
+    _check_inference(bb, relax.op.add(x4, y3), relax.TensorStructInfo(dtype="float32", ndim=-1))
+
+
+def test_infer_struct_info_binary_cmp():
+    bb = relax.BlockBuilder()
+    x = relax.Var("x", R.Tensor((2, 3), "float32"))
+    y0 = relax.Var("y", R.Tensor((2, 3), "float32"))
+    y1 = relax.Var("y", R.Tensor((2, 3), "int32"))
+    _check_inference(bb, relax.op.less(x, y0), relax.TensorStructInfo((2, 3), "bool"))
+    _check_inference(bb, relax.op.less(x, y1), relax.TensorStructInfo((2, 3), "bool"))
+
+
+def test_infer_struct_info_binary_symbolic():
+    bb = relax.BlockBuilder()
+    m = tir.Var("m", "int64")
+    n = tir.Var("n", "int64")
+    k = tir.Var("k", "int64")
+    x0 = relax.Var("x", R.Tensor((m, n), "float32"))
+    x1 = relax.Var("x", R.Tensor((1, n), "float32"))
+    x2 = relax.Var("x", R.Tensor((k, n, m), "float32"))
+    x3 = relax.Var("x", R.Tensor((3, 1, n), "float32"))
+    x4 = relax.Var("x", R.Tensor("float32", ndim=2))
+    y0 = relax.Var("y", R.Tensor((m, n), "float32"))
+    y1 = relax.Var("y", R.Tensor((m, n + 2), "float32"))
```

Comment:
add a few cases with float64, and int8 dtype. Mainly because float32 is the default and to prevent an impl that always return float32